### DISCUSSION TITLE
Running the game and lobby server under a path in an express app

### DISCUSSION THREAD

**Discussion by customautosys** [boardgameio/boardgame.io#1176] [Q&A]
Hi, I'm using Quasar Framework SSR https://quasar.dev/quasar-cli-vite/developing-ssr/ssr-middleware. It uses Express middleware.

Does anyone know if it is possible to run the koa app as an Express middleware on a path within my existing app? I read here that it's possible to just use app.callback(): https://github.com/koajs/koa/issues/846

Is it possible to start the game and lobby server without a port? I do not need it to listen on any port because the requests should just be passed through from Express.

**Comment by fladrif** [comment]
So it looks like the game server is run as a koa server, not a middleware. If you can't adapt your custom code to use koa middlewares and implement that as part of boardgame.io's server, you could potentially just run two servers? My brief overview of quasar is that it's a frontend framework that could just be run independently from the boardgame.io servers, does that sound right?

**Reply by customautosys** [reply]
Quasar is both a frontend framework as well as a server side rendering framework (think replacement for Nuxt), which uses express at the backend.

If anybody has any experience running the boardgame server as a middleware route in an express app do share, that would be really useful for me.

**Reply by fladrif** [reply]
You could, but it's not the recommended way to utilize boardgame.io, nor would I imagine it would be the easiest way to do whatever you want to implement. The `Server()` function from `boardgame.io/server` exposes the koa app object that you could use as an Express application middleware, but you would need to reimplement most of the server functionality, look at the Server code [here](https://github.com/boardgameio/boardgame.io/blob/main/src/server/index.ts#L130) to see what it all does. 

What are you trying to accomplish?
